### PR TITLE
Feature/ifo fixes

### PR DESCRIPTION
**Pull request by hachiojidev**
- Refactor IFO including new designs
- Fix state not updated when contributing

### CODE REVIEW

File: src/hooks/useApprove.ts [hunk 1]
```diff
@@ -67,14 +67,8 @@ export const useLotteryApprove = () => {
...
-    try {
-      const tx = await tokenContract.methods
-        .approve(spenderAddress, ethers.constants.MaxUint256)
-        .send({ from: account })
-      return tx
-    } catch {
-      return false
-    }
+    const tx = await tokenContract.methods.approve(spenderAddress, ethers.constants.MaxUint256).send({ from: account })
+    return tx
```
Comment (hachiojidev) [inline src/hooks/useApprove.ts:71]: Removed try/catch so the error would not be swallowed

File: src/hooks/useAllowance.ts [hunk 2]
```diff
@@ -29,17 +29,17 @@ export const useLotteryAllowance = () => {
...
+  const [allowance, setAllowance] = useState(new BigNumber(0))
 
   useEffect(() => {
     const fetch = async () => {
       try {
         const res = await tokenContract.methods.allowance(account, spenderAddress).call()
         setAllowance(new BigNumber(res))
       } catch (e) {
-        setAllowance(null)
+        console.error(e)
```
Comment (RabbitDoge) [inline src/hooks/useAllowance.ts:42]: Don't you want to display a ui message here ?

File: src/hooks/useApprove.ts (hunk 1 above)
Comment (RabbitDoge) [reply src/hooks/useApprove.ts:71]: The problem was the lack of ui feedback, imo we should keep the try catch and display a notification saying the approval has failed

Comment (hachiojidev) [reply src/hooks/useApprove.ts:71]: This solves that exact problem. It was swallowing the error not allowing the UI to catch it.

Comment (hachiojidev) [reply src/hooks/useApprove.ts:71]: Perhaps we should re-throw the error instead.

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**